src/main/java/com/mercadopago/client/oauth/OauthClient.java

```diff
@@ -0,0 +1,161 @@
+      throws MPException {
+    return this.createCredential(authorizationCode, redirectUri, null);
+  }
+
+  /**
+   * Create Oauth credentials to operate on behalf of a seller. Go <a
+   * href=https://www.mercadopago.com.br/developers/en/guides/security/oauth>here</a> to learn more.
+   *
+   * @param authorizationCode authorization code received from calling getAuthorizationURL
+   * @param redirectUri the redirectUri received from calling getAuthorizationURL
+   * @param requestOptions metadata to customize the request
+   * @return the Oauth credentials
+   */
+  public OauthCredential createCredential(
+      String authorizationCode, String redirectUri, MPRequestOptions requestOptions)
+      throws MPException {
+    CreateOauthCredentialRequest request =
+        CreateOauthCredentialRequest.builder()
+            .clientSecret(getAccessToken(requestOptions))
+            .code(authorizationCode)
+            .redirectUri(redirectUri)
+            .build();
+    IdempotentRequest idempotentRequest =
+        IdempotentRequest.buildRequest(
+            path, HttpMethod.POST, Serializer.serializeToJson(request), null, requestOptions);
+    MPResponse response = send(idempotentRequest);
+
+    OauthCredential credential =
+        Serializer.deserializeFromJson(OauthCredential.class, response.getContent());
+    credential.setResponse(response);
+
+    return credential;
+  }
+
+  /**
+   * Refresh Oauth credentials.
+   *
+   * @param refreshToken refresh token received when you create credentials
+   * @return new Oauth credentials
+   * @throws MPException any error retrieving Oauth credentials
+   */
+  public OauthCredential refreshCredential(String refreshToken) throws MPException {
+    return this.refreshCredential(refreshToken, null);
+  }
+
+  /**
+   * Refresh Oauth credentials.
+   *
+   * @param refreshToken refresh token received when you create credentials
+   * @param requestOptions metadata to customize the request
+   * @return new Oauth credentials
+   * @throws MPException any error retrieving Oauth credentials
+   */
+  public OauthCredential refreshCredential(String refreshToken, MPRequestOptions requestOptions)
+      throws MPException {
+    RefreshOauthCredentialRequest request =
+        RefreshOauthCredentialRequest.builder()
+            .clientSecret(getAccessToken(requestOptions))
+            .refreshToken(refreshToken)
+            .build();
+
+    IdempotentRequest idempotentRequest =
+        IdempotentRequest.buildRequest(
+            path, HttpMethod.POST, Serializer.serializeToJson(request), null, requestOptions);
+    MPResponse response = send(idempotentRequest);
+    OauthCredential credential =
+        Serializer.deserializeFromJson(OauthCredential.class, response.getContent());
+    credential.setResponse(response);
+
+    return credential;
+  }
+
+  private String getAccessToken(MPRequestOptions requestOptions) {
```

Comment:
Acredito que este método vai ser usado em todos os clients, não?
Se for o caso, seria melhor esse método estar em `MercadoPagoClient`.